src/imperative/imperative_utils.h

```diff
@@ -117,11 +117,14 @@ inline void SetShapeType(const Context& ctx,
   for (auto& i : outputs) {
     out_shapes.push_back(i->shape());
   }
-  CHECK(infershape.count(attrs.op))
-    << "Operator " << attrs.op->name << " is missing FInferShape attribute";
-  CHECK(infershape[attrs.op](attrs, &in_shapes, &out_shapes));
-  CHECK_EQ(out_shapes.size(), outputs.size());
-
+  const bool dynamic_shape_exists = [&]() {
```

Comment:
Not a good name here. The major job here is shape inference, and check if dynamic shape exist.

Reply:
@ZhennanQin Thanks for the comment! Could you kindly suggest me a better name?

Reply:
Why not moving shape inference out of this lambda function? Actually, I don't see any reason that using lambda function here.

Reply:
@ZhennanQin personal coding style though: in principle, having stuff defined and constant without uninitialized value helps reduce bugs. Meanwhile, compiler could optimize out the lambda function, so it doesn't introduce any overhead. But I admit that it does not look that straight-forward. So, if you don't like it, I could definitely change it back.